wgpu/src/lib.rs

```diff
@@ -1936,12 +1931,29 @@ impl Instance {
     /// - On macOS/Metal: will panic if not called on the main thread.
     /// - On web: will panic if the `raw_window_handle` does not properly refer to a
     ///   canvas element.
-    pub unsafe fn create_surface<
+    pub fn create_surface<
+        'window,
+        W: raw_window_handle::HasWindowHandle + raw_window_handle::HasDisplayHandle,
+    >(
+        &self,
+        window: &'window W,
+    ) -> Result<Surface<'window>, CreateSurfaceError> {
+        unsafe { self.create_surface_from_raw(window) }
+    }
```

Comment:
So the thing I want to avoid is having surface logically contain a _reference_ to the window. This prevents people from storing both the window and the surface in the same struct (such a struct would be self-referential). The design I proposed stores a W directly. This allows a user to not need to deal with lifetimes in most cases (only if they give us a lifetime-bearing W) and you can easily slot in a Arc<Window>.

Reply:
Ah, I see! That totally makes sense.
Will do that!

Reply:
Done in https://github.com/gfx-rs/wgpu/pull/4597/commits/3d57ebadabb1e356eeebed07d0d5634cdffb3cec.